app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java

```diff
@@ -92,56 +106,43 @@ private void executeGrpcTask(final AbstractGrpcTaskExecutor task) {
             return;
         }
         enableButtons(false);
-        new GrpcTask(this, host, Integer.parseInt(port)).execute(task);
+        new GrpcTask(host, Integer.parseInt(port)).execute(task);
     }
 
-    private void enableButtons(boolean isEnabled) {
+    private void enableButtons(final boolean isEnabled) {
         mMoveForwardButton.setEnabled(isEnabled);
         mInfoButton.setEnabled(isEnabled);
         mReadEncodersButton.setEnabled(isEnabled);
     }
 
-    @Override
-    public void onGrpcTaskCompleted() {
-        enableButtons(true);
-    }
-
     public class GrpcTask extends AsyncTask<AbstractGrpcTaskExecutor, Void, String> {
-        private final String mHost;
-        private final int mPort;
-        private ManagedChannel mChannel;
-        private RoverServiceGrpc.RoverServiceBlockingStub mStub;
-        private IOnGrpcTaskCompleted mListener;
-
-        public GrpcTask(IOnGrpcTaskCompleted listener, String host, int port) {
-            this.mHost = host;
-            this.mPort = port;
-            this.mListener = listener;
-        }
 
         @Override
         protected void onPreExecute() {
-            mChannel = ManagedChannelBuilder.forAddress(mHost, mPort)
-                    .usePlaintext(true)
-                    .build();
-            mStub = RoverServiceGrpc.newBlockingStub(mChannel);
+            super.onPreExecute();
+            enableButtons(Boolean.FALSE);
+        }
+
+        public GrpcTask(final String host, final int port) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Constructor definition in wrong order.](https://www.codacy.com/app/dot-doom/rover-android-client/file/2661510811/issues/source?bid=3324652&fileBranchId=3330575#l126)